### PR TITLE
mysql-server: initscript: allow spaces in 'mysql-config' datadir

### PR DESCRIPTION
**Pull request by bittorf**
also convert script to OpenWrt-style and use
logger instead of echo out to STDERR.

Signed-off-by: Bastian Bittorf <bb@npl.de>

### CODE REVIEW

File: utils/mysql/files/mysqld.init [hunk 1]
```diff
@@ -7,24 +7,31 @@ STOP=10
...
-	local datadir=$(sed -n -e "s/^[[:space:]]*datadir[[:space:]]*=[[:space:]\"']*\([^[:space:]\"']*\)[[:space:]\"']*/\1/p" /etc/my.cnf)
-	if [ ! -d "$datadir" ]; then
-		error "Error: datadir '$datadir' in /etc/my.cnf doesn't exist"
+	local datadir
+	local config='/etc/my.cnf'
+
+	# e.g.: datadir  = '/my/path/with spaces'
+	set -- $( grep ^'datadir' "$config" )
+	shift 2
+	eval datadir="$*"
```
Comment (jow-) [inline utils/mysql/files/mysqld.init]: This breaks unquoted paths with spaces:

    # (set -x; set -- $( echo 'datadir = foo bar' ); shift 2; eval datadir="$*")
    ++ echo 'datadir = foo bar'
    + set -- datadir = foo bar
    + shift 2
    + eval 'datadir=foo bar'
    ++ datadir=foo
    ++ bar
    bash: bar: command not found

It is also still prone to shell injections:

    # (set -x; set -- $( echo 'datadir = $(date)' ); shift 2; eval datadir="$*")
    ++ echo 'datadir = $(date)'
    + set -- datadir = '$(date)'
    + shift 2
    + eval 'datadir=$(date)'
    +++ date
    ++ datadir='Do 1. Dez 10:10:27 CET 2016'

Whats wrong with the propsed sed command? It is safer, requires less forks, does not rely on eval and handles all corner cases nicely.

**Comment by bittorf** [comment]
sorry, will update. the sed-command is just ugly and totally unreadable. i will think about it.

**Comment by jow-** [comment]
It might be ugly but the nice looking code you propose as replacement is full of holes :)

**Comment by bittorf** [comment]
sorry, i learned a lot: see https://github.com/openwrt/packages/pull/3603

**Pull request closed without merging**